GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -229,34 +229,36 @@ void StartAsyncDataLoad()
             {
                 var cancellationToken = _asyncLoadCancellation.Next();
 
-                ThreadHelper.JoinableTaskFactory.RunAsync(() => LoadLinksForRevisionAsync(_revision)).FileAndForget();
-
-                if (_refsOrderDict == null)
+                ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
                 {
-                    ThreadHelper.JoinableTaskFactory.RunAsync(() => LoadSortedRefsAsync()).FileAndForget();
-                }
+                    await LoadLinksForRevisionAsync(_revision);
 
-                // No branch/tag data for artificial commands
+                    if (_refsOrderDict == null)
```

Comment:
`_refsOrderDict ` should be wrapped in AsyncLazy and not called here at all.

Reply:
Perhaps you're right, however it seems like a bigger change than I can afford allocate time for.
If it is left as-is for now we could query refs few extra times, anything else?

Reply:
>anything else?

No, only that.

Reply:
I created a PR for this: https://github.com/gitextensions/gitextensions/pull/6335
It turned out that once loaded the sortedRefs were never refreshed on repository change.